dask/dataframe/shuffle.py

```diff
@@ -346,6 +346,13 @@ def shuffle(
 
     if not isinstance(index, _Frame):
         index = df._select_columns_or_index(index)
+    elif hasattr(index, "to_frame"):
+        # If this is an index, we should still convert to a
+        # DataFrame. Otherwise, the hashed values of a column
+        # selection will not match (important when merging).
+        _divisions = index.divisions
+        index = index.to_frame()
```

Comment:
Does `to_frame()` not preserve divisions?

Reply:
Thanks for calling this out.  Divisions are not preserved for Index -> DataFrame, but I cannot think of a good reason why not.  Perhaps the "correct" fix for this is in `to_frame`

Reply:
Okay - 2069d8c made the necessary change to preserve divisions in `Index.to_frame()`.  Hopefully that makes as much sense as I think it does.

Reply:
cool!